src/plugins/ffmemless/plugin.c

```diff
@@ -572,7 +572,9 @@ static int ffm_play(struct ffm_effect_data *data, int play)
 		}
 		N_DEBUG (LOG_CAT "Starting playback %d", data->id);
 	} else {
-		ffm_playback_done(data);
+		if (ffm.cache_effects) {
```

Comment:
No idea whether this makes sense, but ...

What I find suspicious is the whole data->cached_effect thingy, especially the custom_data part that is a pointer. It is used for storing address in stack. Which becomes invalid as soon as we return from here. And on a quick glance these structures are retained and copied bitwise back and forth -> is there any guarantee that data pointers are / remain valid? (also sometimes custom_len is assigned along with custom_data and sometimes it is not which seems suspicious too)